spring-integration-core/src/main/java/org/springframework/integration/router/AbstractMappingMessageRouter.java

```diff
@@ -173,11 +172,13 @@ public void replaceChannelMappings(Properties channelMappings) {
 	}
 
 	private void doSetChannelMappings(Map<String, String> newChannelMappings) {
-		Map<String, String> oldChannelMappings = this.channelMappings;
-		this.channelMappings = newChannelMappings;
+		Map<String, String> oldChannelMappings = new HashMap<String, String>(this.channelMappings);
+		synchronized (this.channelMappings) {
```

Comment:
M-m-m. OK.
So, looks like I didn't understand the approach for the first time.
Will redo.